### PR TITLE
Added error handle for 401 Error, allow redirect to login auth url

### PR DESCRIPTION
**Pull request by Pockylee**
## Description

As the title, however, I currently haven't found a method to reproduce the error. As a result, we might have to merge and then test in prod.

## Add

Add a third option when the user faces the 401 Error, allow redirect to login #195 issue in pearai-app

### CODE REVIEW

**Comment by Pockylee** [comment]
[Link to issue](https://github.com/trypear/pearai-app/issues/195)

File: extensions/vscode/src/webviewProtocol.ts [hunk 1]
```diff
@@ -118,7 +118,33 @@ export class VsCodeWebviewProtocol {
...
+                  );
+                } else if (selection === 'Troubleshooting') {
+                  vscode.env.openExternal(
+                    vscode.Uri.parse('https://trypear.ai/troubleshooting'),
+                  );
+                } else if (selection === 'Re-login') {
+                  // Redirect to auth login URL
+                  vscode.env.openExternal(
+                    vscode.Uri.parse(
+                      'https://trypear.ai/signin?...llback=pearai://pearai.pearai/auth',
```
Comment (Ellahinator) [inline extensions/vscode/src/webviewProtocol.ts:142]: typo
```suggestion
                      'https://trypear.ai/signin?callback=pearai://pearai.pearai/auth',
```

Comment (Pockylee) [reply extensions/vscode/src/webviewProtocol.ts]: Fixed, tysm

**Comment by nang-dev** [comment]
F on test on prod, nw ill watch it with you

File: extensions/vscode/src/webviewProtocol.ts [hunk 2]
```diff
@@ -118,7 +118,33 @@ export class VsCodeWebviewProtocol {
           );
 
           let message = e.message;
-          if (e.cause) {
+          if (e.status === 401) {
```
Comment (siddharthjha387) [inline extensions/vscode/src/webviewProtocol.ts:121]: Do not use magic number, use constant for '401'. Check at other places also

File: extensions/vscode/src/webviewProtocol.ts [hunk 3]
```diff
@@ -118,7 +118,33 @@ export class VsCodeWebviewProtocol {
...
+                  );
+                } else if (selection === 'Re-login') {
+                  // Redirect to auth login URL
+                  vscode.env.openExternal(
+                    vscode.Uri.parse(
+                      'https://trypear.ai/signin?callback=pearai://pearai.pearai/auth',
+                    ),
+                  );
+                }
+              });
```
Comment (siddharthjha387) [inline extensions/vscode/src/webviewProtocol.ts:146]: Can we wrap this logic inside function?

**Review by siddharthjha387 — CHANGES_REQUESTED** [review]

**Comment by nang-dev** [comment]
imma test this :D

**Comment by nang-dev** [comment]
Added polish and merged!

**Review by nang-dev — APPROVED** [review]

**Comment by Pockylee** [comment]
Thanks for the help!!!

**Pull request merged**